nav2_smac_planner/test/test_node_lattice.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021, Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include "nav2_smac_planner/node_lattice.hpp"
+#include "gtest/gtest.h"
+#include <fstream>
+
+using json = nlohmann::json;
+
+TEST(ParserTest, test_lattice_node)
+{
+    //TODO: Where should the lattice file be placed? 
+    std::string filePath = "/home/josh/nav2_ws/src/navigation2/nav2_smac_planner/test/output.json";
```

Comment:
if its in nav2_smac_planner, you can use the ament to find the packages' shared path https://docs.ros2.org/foxy/api/ament_index_cpp/ if you install that output file somewhere it can retrieve it

Reply:
@jwallace42 by the way, you can find an example of this in `nav2_bt_navigator/src/navigators/navigate_to_pose.cpp` using `ament_index_cpp` so you don't have to search around. I was very happy to have found that API myself earlier this year.

Reply:
Thanks for the example! I am happy you found it too!

Reply:
Done